cmd/mysterium_server/command_run/identity_handler_test.go

```diff
@@ -0,0 +1,48 @@
+package command_run
+
+import (
+	"testing"
+
+	"github.com/mysterium/node/identity"
+	"github.com/mysterium/node/server"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_identityHandler_UseExisting(t *testing.T) {
+	identityManager := identity.NewIdentityManagerFake()
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseExisting("address")
+	assert.Equal(t, identityManager.FakeIdentity1, id)
+	assert.Nil(t, err)
+}
+
+func Test_identityHandler_UseLast(t *testing.T) {
+	identityManager := identity.NewIdentityManagerFake()
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	fakeIdentity := identity.FromAddress("abc")
+	cache.StoreIdentity(fakeIdentity)
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseLast()
+	assert.Equal(t, fakeIdentity, id)
+	assert.Nil(t, err)
+}
+
+func Test_identityHandler_UseNew(t *testing.T) {
+	identityManager := identity.NewIdentityManagerFake()
+	client := server.NewClientFake()
+	cache := identity.NewIdentityCacheFake()
+
+	handler := NewNodeIdentityHandler(identityManager, client, cache)
+
+	id, err := handler.UseNew()
+	assert.Equal(t, identityManager.FakeIdentity2, id)
```

Comment:
You should assert if it was registered via `client`

Reply:
Done.